### DISCUSSION TITLE
Default settings override for react-hook-form?

### DISCUSSION THREAD

**Discussion by mrlubos** [react-hook-form/react-hook-form#2647] [Q&A]
Hi!

Is there a way to override the default settings in `useForm()`? Specifically, I'd like to change the `reValidateMode` attribute for all instances. I am currently creating a wrapper around `useForm()`, but this causes me to lose type annotations.

My question is:
1. Can I globally override the default settings?
2. Can I "inherit" TypeScript annotations from `useForm()`?

Thank you!

**Comment by bluebill1049** [comment]
no, we don't have that, but you can probably set a global `prop` and pass to all your children with useForm.

**Reply by mrlubos** [reply]
Hi @bluebill1049, that would rely on the developer knowing about this `prop` though, right?

**Reply by mrlubos** [reply]
@bluebill1049 An alternative we can do in the meantime is create default config

```tsx
interface FormConfig {
  reValidateMode: Exclude<Mode, 'onTouched' | 'all'>;
}

const formConfig: FormConfig = { reValidateMode: 'onSubmit' };
```

And then pass it to all form instances

```tsx
const form = useForm({
  // some other stuff
  ...config
})
```

It's a bit brittle though because I had to copy-paste the type definition from `react-hook-form` as it's not exported
